### ISSUE TITLE
IndexError: index 3 is out of bounds for axis 0 with size 3

### ISSUE THREAD

**Issue by fandy11638104** [YunYang1994/tensorflow-yolov3#309]
![image](https://user-images.githubusercontent.com/54304106/64306060-1555e080-cfc4-11e9-9af7-1f5c562365ae.png)

**Comment by fandy11638104** [comment]
问题已经解决，是voc.names与xml不相符的问题

**Comment by Byronnar** [comment]
> 问题已经解决，是voc.names与xml不相符的问题

请问你是怎么不相符？我也出现这个情况了

**Comment by 3d2ydym** [comment]
> 问题已经解决，是voc.names与xml不相符的问题

请问是怎么解决的，我用的是coco 数据集，也遇到了这样的问题


  File "train.py", line 184, in <module>
    if __name__ == '__main__': YoloTrain().train()
  File "train.py", line 144, in train
    for train_data in pbar:
  File "C:\Users\22673\AppData\Roaming\Python\Python36\site-packages\tqdm\std.py", line 1081, in __iter__
    for obj in iterable:
  File "E:\ai\tensorflow-yolov3\core\dataset.py", line 82, in __next__
    label_sbbox, label_mbbox, label_lbbox, sbboxes, mbboxes, lbboxes = self.preprocess_true_boxes(bboxes)
  File "E:\ai\tensorflow-yolov3\core\dataset.py", line 247, in preprocess_true_boxes
    label[best_detect][yind, xind, best_anchor, :] = 0
IndexError: index 63 is out of bounds for axis 1 with size 60

**Comment by MalikFareed** [comment]
check your classes numbers(in 'voc_test.txt' or 'voc_train.txt') if their index are not out of classes mentioned in 'coco.names'. I solved this problem like this.

**Comment by fandy11638104** [comment]
thank you !



At 2019-11-11 02:16:51, "MalikFareed" <notifications@github.com> wrote:

check your classes numbers(in 'voc_test.txt' or 'voc_train.txt') if their index are not out of classes mentioned in 'coco.names'. I solved this problem like this.

**Comment by MalikFareed** [comment]
Yes i resolved this months ago, like you mentioned. Btw thanks alot for
helping.
